test/dynamo/test_modules.py

```diff
@@ -2682,6 +2685,34 @@ def fn(x):
         self.assertEqual(test_functions._variable, 1)
         self.assertEqual(res, 3 * torch.ones(10))
 
+
+    def test_save_and_load1(self):
+        mod = MockModule()
+        opt_mod = torch.compile(mod)
+        inp = torch.randn(10, 10)
+        opt_mod(inp)
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            torch.save(opt_mod, os.path.join(tmpdirname, "model.pt"))
+            loaded_model = torch.load(os.path.join(tmpdirname, "model.pt"))
+            loaded_model(inp)
+            self.assertTrue(same_two_models(loaded_model, mod, [inp]))
+            self.assertTrue(same_two_models(loaded_model, opt_mod, [inp]))
```

Comment:
Can you add a test that the loaded model is actually compiled?  (rather than running in eager)

Reply:
I could only check whether an 'OptimizedModule' the loaded model is. 
I've no idea what a compiled model would be. 
Maybe I would check the difference of the weight between the loaded model and the opt_mod.
As far as I know, the model will be compiled just before the first running.
So are you meaning I should check the weight after checking the answer?
If so, there is a question whether the weight should be the same after compiled once or more.
Or just whether an 'OptimizedModule' the loaded model is?

Reply:
You could do something like:
```
torch._dynamo.reset()  # force recompiles
torch._inductor.metrics.generated_kernel_count = 0
loaded_model(inp)
self.assertGreater(torch._inductor.metrics.generated_kernel_count, 0)
```